core/src/main/scala/org/apache/spark/shuffle/ShuffleManager.scala

```diff
@@ -54,6 +54,19 @@ private[spark] trait ShuffleManager {
       context: TaskContext,
       metrics: ShuffleReadMetricsReporter): ShuffleReader[K, C]
 
+  /**
+   * Get a reader for a range of reduce partitions (startPartition to endPartition-1, inclusive) to
+   * read from mapId.
+   * Called on executors by reduce tasks.
+   */
+  def getMapReader[K, C](
+      handle: ShuffleHandle,
+      startPartition: Int,
+      endPartition: Int,
```

Comment:
now the local shuffle reader always read all the shuffle blocks of one mapper. Let's add the above 2 parameters later when we going to use them.